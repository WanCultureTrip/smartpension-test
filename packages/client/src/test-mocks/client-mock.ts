import type { MockedResponse } from '@apollo/client/testing'
import { CITY_FILTER } from '../graphql/CITY_FILTER.query'
import { UPDATE_CITY } from '../graphql/UPDATE_CITY.query'
import type { CityType } from '../types/city-types'

const mockCityList = [
  {
    id: 1,
    name: 'London',
    visited: true,
    wishlist: true,
    country: 'United Kingdom',
  },
  {
    id: 14,
    name: 'Barcelona',
    visited: true,
    wishlist: false,
    country: 'Spain',
  },
  {
    id: 50,
    name: 'Lodz',
    visited: false,
    wishlist: false,
    country: 'Poland',
  },
  {
    id: 82,
    name: 'Oslo',
    visited: false,
    wishlist: false,
    country: 'Norway',
  },
]

const getMockResults = (cityList: CityType[]) => {
  return {
    data: {
      cities: {
        total: cityList.length,
        cities: cityList,
      },
    },
  }
}

export const clientMock = [
  {
    request: {
      query: CITY_FILTER,
      variables: {
        filter: { name: 'Lo' },
      },
    },
    result: getMockResults(mockCityList),
  },
  {
    request: {
      query: UPDATE_CITY,
      variables: {
        input: {
          visited: true,
          wishlist: false,
          id: 82,
        },
      },
    },
    result: { data: { updateCity: { ...mockCityList[3], visited: true } } },
  },
  {
    request: {
      query: CITY_FILTER,
      variables: {
        filter: { visited: true },
      },
    },
    result: getMockResults(mockCityList.filter(item => item.visited)),
  },
  {
    request: {
      query: CITY_FILTER,
      variables: {
        filter: { wishlist: true },
      },
    },
    result: getMockResults(mockCityList.filter(item => item.wishlist)),
  },
] as MockedResponse[]
