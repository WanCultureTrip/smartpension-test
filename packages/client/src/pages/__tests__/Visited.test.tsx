import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Visited } from '../Visited'
import { MockedProvider } from '@apollo/client/testing'
import { clientMock } from '../../test-mocks/client-mock'

const renderComp = () =>
  render(
    <MockedProvider mocks={clientMock} addTypename={false}>
      <Visited />
    </MockedProvider>
  )

describe('<Visited /> component', () => {
  test('should match initial snapshot', async () => {
    const { container } = renderComp()
    expect(container).toMatchSnapshot()
  })

  test('should display visited cities', async () => {
    const { getAllByLabelText, findByText } = renderComp()

    // wait for cards to load
    await waitFor(() => getAllByLabelText('city-card'))

    // check that 2 city cards is loaded
    expect(getAllByLabelText('city-card').length).toBe(2)

    // check that one of the cities is London
    expect(findByText('London')).toBeDefined()
  })
})
