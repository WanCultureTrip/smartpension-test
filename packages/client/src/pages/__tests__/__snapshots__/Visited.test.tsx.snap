// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Visited /> component should match initial snapshot 1`] = `
<div>
  <h1
    class="chakra-heading css-0"
  >
    Visited
  </h1>
  <div
    class="chakra-container css-gygvqc"
  >
    <div
      class="css-90haw7"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="css-1demtxk"
        data-indeterminate=""
        role="progressbar"
        style="width: 0%;"
      />
    </div>
    <div
      aria-label="city-list"
      class="css-63tcgz"
    />
  </div>
</div>
`;
